dev/tests/agent.rs

```diff
@@ -488,3 +495,53 @@ async fn handle_agent_restart() -> Result<()> {
 
     Ok(())
 }
+
+#[integration_test]
+async fn handle_termination_request() -> Result<()> {
+    let nats = Nats::new().await?;
+    let agent = Agent::new(&nats).await?;
+    let nats = nats.connection().await?;
+    let mut controller_mock = MockController::new(nats.clone()).await?;
+
+    let mut request = base_spawn_request();
+
+    controller_mock
+        .expect_handshake(request.drone_id, agent.ip)
+        .await?;
+    controller_mock
+        .expect_status_message(request.drone_id, "spawner.test")
+        .await?;
+
+    request.max_idle_secs = Duration::from_secs(1000);
+    let mut state_subscription = BackendStateSubscription::new(&nats, &request.backend_id).await?;
+
+    controller_mock.spawn_backend(&request).await?;
+    state_subscription
+        .wait_for_state(BackendState::Ready, 60_000)
+        .await?;
+
+    let proxy_route = agent
+        .db
+        .get_proxy_route(request.backend_id.id())
+        .await?
+        .expect("Expected proxy route.");
+    tokio::spawn(async move {
```

Comment:
I guess this is just to keep it alive? Instead, let's just use a long grace period.

I've also created an issue (#128) to allow more flexible termination conditions, including no maximum idle time, which we could use here once implemented.

Reply:
hehe yes.